x-pack/auditbeat/processors/sessionmd/add_session_metadata.go

```diff
@@ -56,33 +58,35 @@ func New(cfg *cfg.C) (beat.Processor, error) {
 		return nil, fmt.Errorf("failed to create DB: %w", err)
 	}
 
-	backfilledPIDs := db.ScrapeProcfs()
-	logger.Infof("backfilled %d processes", len(backfilledPIDs))
+	if c.Backend != "modern" {
+		backfilledPIDs := db.ScrapeProcfs()
+		logger.Infof("backfilled %d processes", len(backfilledPIDs))
+	}
 
 	var p provider.Provider
 
 	switch c.Backend {
 	case "auto":
-		p, err = ebpf_provider.NewProvider(ctx, logger, db)
+		p, err = modernprovider.NewProvider(ctx, logger)
```

Comment:
The `ctx` that we are passing in should be made cancelable such that when the processor is `Close()`ed that context gets canceled. Then any resources tied into it can be released.